### DISCUSSION TITLE
items are showing last update time every 5 seconds

### DISCUSSION THREAD

**Discussion by tragicdog** [swiftbar/SwiftBar#188] [Q&A] [answered]
As the title says my plug-ins (some I've written most I haven't) are showing the date/time every 5 seconds. It's made the app almost unusable as it's very distracting. 

<img width="653" alt="Screen Shot 2021-04-20 at 6 06 20 PM" src="https://user-images.githubusercontent.com/322642/115482279-29833b80-a203-11eb-8ed2-134027d79002.png">

Anyone have any ideas?

**Comment by melonamin** [comment]
A couple of questions:
- Version of SwiftBar?
- Can you share a plugin which gives you a date in the menu bar?

**Reply by tragicdog** [reply]
version 1.2.3

Linode Status by Brendon Cheves
Homebrew Updates by killercup
Hacker News Headlines - Lite by amrrs

Currently those are the only plug-ins I'm running and they all rotate between the date/time of last update and what hey are supposed to show.

**Reply by melonamin** [reply]
Tested the Homebrew plugin - works great, so it's not the plugin.

Do you mind to give a try to 1.3.0 and [Debug Plugin](https://github.com/swiftbar/SwiftBar/issues/182) feature? It will give us a look at plugin output exactly like SwiftBar sees it.

**Reply by tragicdog** [reply]
done and done. Here's the home-brew plug-in. 
<img width="612" alt="Screen Shot 2021-04-21 at 12 56 54 PM" src="https://user-images.githubusercontent.com/322642/115613307-18890780-a2a1-11eb-8978-2d8d2d5c02e7.png">

And what is showing in the menu bar every 5 seconds. 
<img width="263" alt="Screen Shot 2021-04-21 at 12 57 32 PM" src="https://user-images.githubusercontent.com/322642/115613379-2b034100-a2a1-11eb-8e2e-8cfe92042da1.png">

edit 1: In the Debug Window, where does "Print SwiftBar ENV" and "Print Plugin Metadata" show up?

edit 2: if I hit "Refresh Plugin" the time showing in the menu bar changes to reflect that time.

**Reply by melonamin** [reply]
Can you copy full text from Debug view(it is scrollable) and share it here?

**Reply by tragicdog** [reply]
I'm unable to select the text in the window. So here's a screen shot. 😀 
![Screen Shot 2021-04-21 at 1 09 00 PM](https://user-images.githubusercontent.com/322642/115614621-c812a980-a2a2-11eb-86a2-57f3d4f3a282.png)

**Reply by melonamin** [reply]
> I'm unable to select the text in the window
That's cool, you've found a bug! :)

Sorry for back and forth, but can you do a "clean room" experiment:
* Relaunch SwiftBar
* Wait until you see the date-time in the menu bar
* go and ~~copy~~ screenshot debug view, without pressing any buttons on it

Thanks!

**Reply by tragicdog** [reply]
I don't mind the back and forth at all. 

Here is a screen shot of the debug window being opened while the home-brew plugin was showing time and not the number of updates after quitting and relaunching the app. 

<img width="612" alt="Screen Shot 2021-04-21 at 2 34 10 PM" src="https://user-images.githubusercontent.com/322642/115624018-df579400-a2ae-11eb-92de-5d04646ef393.png">

**Reply by melonamin** [reply]
Ok, I feel like we getting closer. This `Wed Apr 21 14:33:53 PDT 2021` is not coming from the plugin, and my guess it is something in your bash env.

To test this theory can you disable bash for this plugin? 
Just add `# <swiftbar.runInBash>false</swiftbar.runInBash>` to the file.

**Reply by tragicdog** [reply]
To the plugin file correct?

**Reply by tragicdog** [reply]
you nailed it. When I ran bash as my shell I had "date" at the end of it to let me know it loaded correctly. Since moving to ZSH that is no longer there, but the file was. The plug-in was creating a new bash env every time it refreshed, and that called the profile. 

removed the line from my bash profile and restarted SwiftBar. 

Thank you for spending your time helping with this. Keep up the good work.

**Comment by tragicdog** [comment, marked as answer]
The plug-ins were calling my bash_profile every time they refreshed that added "date" to the menu bar. Removed that line from my profile and restarted SwiftBar. Alls good.
